torch/_inductor/cudagraph_trees.py

```diff
@@ -1345,9 +1373,14 @@ def try_end_curr_warmup(self):
     def dealloc_current_path_weakrefs(self):
         # TODO: we could also allow the these weak refs to continue to be allocated,
         # but that adds some complications.
-        for t in self.current_node.path_live_weakrefs():
+        for t, stack_trace in self.current_node.path_live_weakrefs_and_stacktraces():
+            # TODO: dont need to test t(), but would need to deduplicate storages
             if t():
                 torch._C._free_And_Remove_DeleterFn(t())
+                warnings.warn(
+                    f"CUDA Graphs is Deallocating Tensor Output from {stack_trace.strip()}. "
+                    "Deallocate output tensor no longer in use or copy output."
```

Comment:
This is not a good warning. Make it better.

Reply:
What would you suggest ?

Reply:
CUDAGraphTree triggered deallocation of tensor {id(t)} due to REASON, even though this tensor is still live. Subsequent use of this tensor may return garbage result.

Also, if we have access to the old Tensor objects, I wonder if we can modify them to make them error if you try to read. This would be much much better than this warning (which is not very actionable).

Reply:
Tensors arent around here, just storages. I agree making storage access fail would be good if it's possible.

Reply:
Well, all I am really asking for here is a warning that is more clear. Imagine someone is debugging a SEV where there is an NE explosion and they're looking through the warnings and they see this warning. Will they be able to figure it out from this warning text? What can help them figure it out?

Reply:
The stack trace and the suggested message (copy the output) is a good start IMO, and gives an actionable message. 

I also think there are other changes to be made that should make the whole thing safer.

Reply:
I haven't the foggiest how I would copy the output. Let's take your test case as an example:

```
        def test_warnings_on_dealloc(self):
            @torch.compile()
            def foo(x):
                return x * x * x

            inp = torch.rand([4], device="cuda")
            out = foo(inp)
            warnings.resetwarnings()
            with warnings.catch_warnings(record=True) as w:
                foo(inp)

            self.assertTrue(len(w) == 1)
            self.assertTrue("x * x * x" in str(w[0]))
```

According to the test, the warning gives you a stack trace that identifies "return x * x * x". So can I fix the problem by saying "return (x * x * x).clone()"? This seems unlikely to help, as the clone will get shoved in the CUDA graph.

Reply:
Copy the output of `torch.compile` is more clear. `tensor {id(t)}` is a temporary value because aot autograd will detach, so I'm not sure what value it has to user

Reply:
```
            @torch.compile()
            def foo(x):
                return x * x * x

            inp = torch.rand([4], device="cuda")
            out = foo(inp).clone()
 ```

Reply:
Don't explain this to me, explain it to the user, in the warning!